src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -154,13 +159,21 @@ class BookingExtractionService(
 
     bookingCalculation.dates[ESED] = latestUnadjustedExpiryDate
     bookingCalculation.effectiveSentenceLength = getEffectiveSentenceLength(
-      booking.getAllExtractableSentences().minOf { it.sentencedAt },
+      earliestSentenceDate,
       latestUnadjustedExpiryDate
     )
 
     return bookingCalculation
   }
 
+  private fun extractManyHomeDetentionCurfewEligibilityDate(earliestSentenceDate: LocalDate, latestUnadjustedExpiryDate: LocalDate, mostRecentSentenceByReleaseDate: ExtractableSentence): LocalDate? {
+    val fourYearSentence = earliestSentenceDate.plusYears(4)
```

Comment:
Might want to replace this magic numbeer at some point?

Reply:
Done